src/DynamoCoreWpf/ViewModels/Search/SelectionNavigator.cs

```diff
@@ -70,12 +72,25 @@ internal void MoveSelection(NavigationDirection direction)
             if (root == null || !root.Any())
                 return;
 
+            if (CurrentlySelection == topResult)
+            {
+                // We can only move forward...
+                if (direction == NavigationDirection.Forward)
+                {
+                    SelectItem(0, 0, 0);
+                    return;
+                }
+                else
+                    // Selected element is top result.
+                    // There is no way to move backward.
+                    return;
+            }
+
             var selectedCategory = root.ElementAt(selectedCategoryIndex);
             var selectedMemberGroup = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex);
 
             // Clear the current selection, no matter what.
-            selection.IsSelected = false;
-            selection = null;
+            CurrentlySelection.IsSelected = false;
```

Comment:
Oops, I didn't notice earlier that the name `CurrentlySelection` is incorrect, it should be `CurrentSelection` (but that's a different thing if it is hard to change right now).